client/python/cryoet_data_portal/src/cryoet_data_portal/_codegen.py

```diff
@@ -0,0 +1,231 @@
+"""This module generates model code from the GraphQL schema."""
+
+import inspect
+import logging
+import pathlib
+import re
+from dataclasses import dataclass
+from textwrap import dedent
+from typing import List, Optional
+
+from graphql import (
+    GraphQLField,
+    GraphQLList,
+    GraphQLNonNull,
+    GraphQLObjectType,
+    GraphQLScalarType,
+    GraphQLSchema,
+    GraphQLType,
+    build_schema,
+)
+
+from cryoet_data_portal import _model_stubs
+
+"""Maps GraphQL field type names to model field defaults and Python types."""
+GQL_TO_MODEL_FIELD = {
+    "Boolean": ("BooleanField()", "bool"),
+    "Float": ("IntField()", "int"),
+    "Int": ("IntField()", "int"),
+    "String": ("StringField()", "str"),
+    "date": ("DateField()", "date"),
+    "numeric": ("FloatField()", "float"),
+    "_numeric": ("StringField()", "str"),
+    "tomogram_type_enum": ("StringField()", "str"),
+}
+
+
+"""Maps GraphQL type names to model class names."""
+GQL_TO_MODEL_TYPE = {
+    "datasets": "Dataset",
+    "dataset_authors": "DatasetAuthor",
+    "dataset_funding": "DatasetFunding",
+    "runs": "Run",
+    "tomogram_voxel_spacings": "TomogramVoxelSpacing",
+    "tomograms": "Tomogram",
+    "tomogram_authors": "TomogramAuthor",
+    "annotations": "Annotation",
+    "annotation_files": "AnnotationFile",
+    "annotation_authors": "AnnotationAuthor",
+    "tiltseries": "TiltSeries",
+}
+
+
+@dataclass(frozen=True)
+class FieldInfo:
+    """Describes the information about a parsed model field."""
+
+    name: str
+    description: str
+    annotation_type: str
+    default_value: str
+
+
+def write_models(path: str) -> None:
+    schema = load_schema()
+    with open(path, "w") as f:
+        f.write(
+            dedent(
+                """\
```

Comment:
I should probably be using a templating tool (e.g. jinja2) instead of all these writes.

We could also consider defining a template per model, which could also be useful for handling the utility methods.